### PR TITLE
[KERNEL32] Fix RaiseException to correctly copy the passed in parameters as ULONG_PTR

### PR DESCRIPTION
**Pull request by tkreuzer**
Fixes C++ exception handling on x64.

### CODE REVIEW

**Comment by JoachimHenze** [comment]
Sounds great, how does your testcase look like, that you used to verify the fixes correctness?

**Comment by tkreuzer** [comment]
> Sounds great, how does your testcase look like, that you used to verify the fixes correctness?

I looked at the thing that I had made, and, behold, it was very good!

**Comment by JoachimHenze** [comment]
> I looked at the thing that I had made, and, behold, it was very good!

Please provide a simple test-case! Reproduction steps of what you used as a specific testcase in words would be enough.

**Review by DarkFire01 — APPROVED** [review]

**Comment by tkreuzer** [comment]
> Please provide a simple test-case! Reproduction steps of what you used as a specific testcase in words would be enough.

You approve shit you don't understand at all, but ask for a testcase when the fix is obvious for a 3 year old. I conclude you are only doing that to piss me off. So what about NO?

**Review by katahiromz — APPROVED** [review]

**Review by learn-more — APPROVED** [review]

File: dll/win32/kernel32/client/except.c [hunk 1]
```diff
@@ -697,14 +697,16 @@ UnhandledExceptionFilter(IN PEXCEPTION_POINTERS ExceptionInfo)
...
+RaiseException(
+    _In_ DWORD dwExceptionCode,
+    _In_ DWORD dwExceptionFlags,
+    _In_ DWORD nNumberOfArguments,
+    _In_opt_ const ULONG_PTR *lpArguments)
 {
     EXCEPTION_RECORD ExceptionRecord;
 
     /* Setup the exception record */
+    RtlZeroMemory(&ExceptionRecord, sizeof(ExceptionRecord));
```
Comment (SergeGautherie) [inline dll/win32/kernel32/client/except.c:709]: Then, `= NULL` and `= 0` lines below could be removed.
Or, you could zero `ExceptionRecord.ExceptionInformation` only.
Is this zeroing even needed at all?

**Comment by JoachimHenze** [comment]
> You approve shit you don't understand at all

I do approve code that can be tested, either manually or automatically.

> I conclude you are only doing that to piss me off. the fix is obvious for a 3 year old.

Yes, your change is obviously an improvement. Even 3 years-old reactosfanboy can see that. But it doesn't tell me whether exception handling for x64 will work after that fix yet, or whether it is just "less broken than it was before, and would need more fixes afterwards". That is what I am trying to find out without digging into the testing here myself. So if you could please answer that question.

File: dll/win32/kernel32/client/except.c [hunk 2]
```diff
@@ -697,14 +697,16 @@ UnhandledExceptionFilter(IN PEXCEPTION_POINTERS ExceptionInfo)
...
     ExceptionRecord.ExceptionCode = dwExceptionCode;
     ExceptionRecord.ExceptionRecord = NULL;
     ExceptionRecord.ExceptionAddress = (PVOID)RaiseException;
```
Comment (HBelusca) [inline dll/win32/kernel32/client/except.c:712]: Is this cast needed by the way?

**Review by HBelusca — APPROVED** [review]

File: dll/win32/kernel32/client/except.c (hunk 1 above)
Comment (JoachimHenze) [reply dll/win32/kernel32/client/except.c:709]: Maybe at least comment the now duplicated assignments then, that Serge pointed to? (my gutfeeling tells me that you do hesitate to delete them entirely)

**Comment by tkreuzer** [comment]
>  But it doesn't tell me whether exception handling for x64 will work after that fix yet, or whether it is just "less broken than it was before, and would need more fixes afterwards". 

No, it doesn't. Neither would a testcase. So you have to take my word for it: It is less broken an still needs more fixes.
But what does that even matter? If you want to know the level of brokenness, you can ask, or if you don't trust my word, grab a debugger and figure it out yourself. This is a fix for broken code, not a "Brief Joachim on the latest state of x64 exception handling" task and especially not a "Provide legal proof for the correctness of x64 exception handling to procure further money from our investors".

**Pull request merged**